src/ensembl/src/content/app/browser/browserActions.ts

```diff
@@ -399,3 +401,43 @@ export const toggleRegionEditorActive = createAction(
 export const toggleRegionFieldActive = createAction(
   'browser/toggle-region-field-active'
 )<boolean>();
+
+export const deleteGenome = createAction('browser/delete-genome')<string>();
+
+export const deleteSpeciesAndSave = (
+  genomeIdToRemove: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    if (activeGenomeId === genomeIdToRemove) {
+      browserStorageService.clearActiveGenomeId();
+    }
+
+    dispatch(deleteGenome(genomeIdToRemove));
+
+    const updatedActiveEnsObjectIds = omit(
+      getBrowserActiveEnsObjectIds(state),
+      genomeIdToRemove
+    );
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+
+    browserStorageService.updateActiveEnsObjectIds({
+      [genomeIdToRemove]: undefined
+    });
+    browserStorageService.updateChrLocation({
+      [genomeIdToRemove]: undefined
+    });
+    browserStorageService.updateTrackPanels({
+      [genomeIdToRemove]: undefined
+    });
+    browserStorageService.saveTrackStates({
+      [genomeIdToRemove]: undefined
+    });
```

Comment:
Just wondering whether we wouldn't prefer to have a single `browserStorageService` method (e.g. `browserStorageService.deleteGenome` that would call all the above four methods.

Reply:
I thought the same but I felt it is better to keep the storage service simpler.

Reply:
It won't complicate the service, but will make the code of the consumer of the service simpler. Deleting a species is practically what in the database world would amount to a transaction. We don't want to expose the complexity of removing a species to the consumer; rather, it's much more preferable to hide the complexity inside the service.

Reply:
That's done